design/open/lakefs-kv-execution-plan.md

```diff
@@ -51,22 +52,23 @@ When a complete data model / functionality completes the transition we can exclu
 
 ## Global changes
 
-- Transition from db to kb should include a configuration flag that enables the use of 'kv' vs the current 'db' implementation.
+- Transition from db to kv should include a configuration flag that enables the use of 'kv' vs the current 'db' implementation.
   As complete features move to use kv, and migration plan is implemented, we can remove the flag check for the feature and just enable the new functionality.
 - Migrate from current DDL to kv will be supported only for Postgres and dropped when all features works using kv.
-- Basic dump/restore functionality should be supported for the k/v implementation, how do we migrate the information - discovery
+- Dump and restore functionality implementation as part of ref-dump.
 
 ## Testing
 
-- Migration - running the same tests on data that was generated with previous db/model
-- Functionality performance - set of tests that will check that we didn't degrade
-- Migration from old format performance - can't take 24h to switch to new version (downtime?)
-- New locking mechanism functionality
+- Migration - upgrade from db to kv version and verify commit log, branch staging environment, authorization and actions data integrity.
+- Functionality - verify that all data kept on kv works with the current functionality (CRUD).
+- Ref-store dump/restore state test that old data is available and not corrupted.
+- Functionality performance - set of tests that will check that we didn't degrade.
+- Migration from old format performance - can't take 24h to switch to new version (downtime?).
+- New locking mechanism functionality. Multiple commits. Read and write while commit. Commit after a failed commit.
 
 ## Milestone
 
-- implement adapter k/v for Postgres
-- feature flag (internal) used for kv development
-- each milestone will include migrate db that will use the new kv format. will support only the part we decided as feature complete and will no longer control by the feature flag.
-- for each step: implement the new models to k/v and migrate code
-- implement adapter k/v for other implementations
+- Implement adapter k/v for Postgres. Unit test. Performance test.
+- Implement authorization, actions and multi part using kv. Include feature flag to control which implementation is active and migrate information to move from db to kv version. Unit test.
+- Implement graveler staging over kv, using kv to manage branch locking for commit and merge. Feature flag. Migrate information from db to kv for graveler (commit log, staging, ref-dump and restore). Unit test. Performance test.
+- Integration testing. Remove old implementation.
```

Comment:
I think we're missing some packages like catalog and diagnostics.. Where do they go?

Reply:
- Catalog just passing the two DB connections to Graveler - so the code will be removed as we remove the 'db' package.
- Diagnostics I'll add a step - we will need to reimplement or drop  the functionality if we like to get the same information from all databases.